core/drivers/imx_i2c.c

```diff
@@ -132,10 +134,28 @@ static void i2c_set_prescaler(uint8_t bid, uint32_t bps)
 
 static void i2c_set_bus_speed(uint8_t bid, int bps)
 {
-	/* Enable the clock */
-	io_write32(i2c_clk.base.va + CCM_CCGRx_SET(i2c_clk.i2c[bid]),
-		   CCM_CCGRx_ALWAYS_ON(0));
-
+	vaddr_t addr = i2c_clk.base.va;
+	uint32_t val = 0;
+
+#if defined(CFG_MX8MM)
```

Comment:
Use of `if (IS_ENABLED(CFG_MX8MM)) { ... }` from `<config.h>` is preferred.

Reply:
ok will fix.
I also need to review some SPL settings which might have been hiding an issue on this PR.

Reply:
about this, unfortunately we cant use it in this instance as the imx6ull uses some definitions that are not defined for mx8mm...ie BM_CCM_CCGR2_I2C1_SERIAL

Reply:
Ah, yes. #ifdef is OK then.